kit/dapp/src/lib/utils/zod/validators/bigint.ts

```
@@ -0,0 +1,127 @@
+/**
+ * BigInt Validation Utilities
+ *
+ * Provides Zod schemas for validating and transforming BigInt values from various input types.
+ * Designed for blockchain applications handling large numeric values like token amounts.
+ *
+ * @module BigIntValidation
+ */
+import { abs, floor, from, multiply, toString, type Numberish } from "dnum";
```

Biome (1.9.4)

[error] Do not shadow the global "toString" property.

Consider renaming this variable. It's easy to confuse the origin of variables when they're named after a known global.

(lint/suspicious/noShadowRestrictedNames)